libgadget/timestep.c

```diff
@@ -314,15 +314,15 @@ do_the_short_range_kick(int i, inttime_t tistart, inttime_t tiend)
         }
 
         /* Code here imposes a hard limit (default to speed of light)
-         * on the gas velocity. Then a limit on the change in entropy
-         * FIXME: This should probably not be needed!*/
-        const double velfac = sqrt(All.cf.a3inv);
+         * on the gas velocity. This should rarely be hit.*/
+        const double velfac = 1./All.cf.a;
         double vv=0;
         for(j=0; j < 3; j++)
             vv += P[i].Vel[j] * P[i].Vel[j];
         vv = sqrt(vv);
 
         if(vv > All.MaxGasVel * velfac) {
```

Comment:
Shouldn't this always be vv * ainv > All.MaxGasVel?
Then in parameters, MaxGasVel shall be defined in terms of peculiar velocity, a dx / dt and thus 3e5 in km/s.

Reply:
Oops, yes it should! Nicely caught.